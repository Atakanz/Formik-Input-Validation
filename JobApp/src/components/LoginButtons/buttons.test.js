import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';
import Buttons from './buttons';

describe('<Buttons />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Buttons />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('works correctly', () => {
    const mockFunction = jest.fn();
    const comp = render(<Buttons onClick={mockFunction} />);
    const buttonTouchable = comp.getByTestId('login-button-title');
    fireEvent(buttonTouchable, 'press');
    expect(mockFunction).toHaveBeenCalled();
  });
});
