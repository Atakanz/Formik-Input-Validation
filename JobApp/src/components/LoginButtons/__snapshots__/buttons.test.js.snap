// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buttons /> renders correctly 1`] = `
<RCTSafeAreaView>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#212121",
        "borderRadius": 20,
        "borderWidth": 1.4,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 5,
        "marginTop": 20,
        "opacity": 1,
        "paddingHorizontal": 50,
        "paddingVertical": 8,
      }
    }
    testID="login-button-title"
  >
    <Text
      style={
        {
          "color": "#212121",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
